server/src/main/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlus.java

```diff
@@ -208,15 +343,21 @@ public void writeTo(long bucket, StreamOutput out) throws IOException {
         out.writeVInt(precision());
         if (algorithm.get(bucket) == LINEAR_COUNTING) {
             out.writeBoolean(LINEAR_COUNTING);
-            lc.bucket = bucket;
-            AbstractLinearCounting.HashesIterator hashes = lc.values();
+            low.bucket = bucket;
+            AbstractLinearCounting.HashesIterator hashes;
+            if (low.exhausted()) {
+                lc.ord = bucketToOrd(bucket);
+                hashes = lc.values();
+            } else {
+                hashes = low.values();
+            }
             out.writeVLong(hashes.size());
             while (hashes.next()) {
                 out.writeInt(hashes.value());
             }
         } else {
             out.writeBoolean(HYPERLOGLOG);
-            hll.bucket = bucket;
+            hll.ord = bucketToOrd(bucket);
```

Comment:
Will we need some wire compatibility stuff here? I actually think maybe not? At first I thought we we would, but maybe that just isn't a thing we need because we're already translating on the "outside".